### PR TITLE
adding code for auth using auth code exchange

### PR DESCRIPTION
**Pull request by nramyasri-sf**

### CODE REVIEW

File: src/authInfo.ts [hunk 1]
```diff
@@ -645,8 +647,10 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
         } else if (!options.authCode && options.refreshToken) {
           // refresh token flow (from sfdxUrl or OAuth refreshFn)
           authConfig = await this.buildRefreshTokenConfig(options);
+        } else if (options.authCode && options.clientId) {
+          authConfig = await this.authorize(options);
```
Comment (tnoonan-salesforce) [inline src/authInfo.ts]: Maybe add a comment that this is to support authCode exchange for access token.

**Comment by tnoonan-salesforce** [comment]
Good job Ramya.

File: src/authInfo.ts [hunk 2]
```diff
@@ -24,6 +25,7 @@ import {
 import { createHash, randomBytes } from 'crypto';
 import * as dns from 'dns';
 import { OAuth2, OAuth2Options, TokenResponse } from 'jsforce';
+import { Connection as JSConnection } from 'jsforce';
```
Comment (rvwatch) [inline src/authInfo.ts]: I'm wondering if it's possible to merge these imports into one line? I think I've seen this before. I could be wrong. 

_example:_
import { OAuth2, OAuth2Options, TokenResponse, Connection as JSConnection  } from 'jsforce';

**Pull request merged**